src/render/processor.rs

```diff
@@ -7,6 +7,18 @@ use crate::SampleRate;
 
 use super::{graph::Node, AudioRenderQuantum, NodeIndex};
 
+#[non_exhaustive] // we may want to add user-provided blobs to this later
+#[derive(Copy, Clone, Debug)]
+/// The execution context of all AudioProcessors in a given AudioContext
+///
+/// This struct currently only contains information about the progress of time. In a future
+/// version, it should be possible to add arbitrary data. For example, multiple processors might
+/// share a buffer defining a wavetable or an impulse response.
+pub struct GlobalScope {
```

Comment:
I organized the three variables in the GlobalScope now, which is loosely based on the https://www.w3.org/TR/webaudio/#audioworkletglobalscope
These concepts of scopes and worklets don't really translate to rust so this is perhaps the best we can do. With the current setup we leave the future ability open (with no breaking changes) for
> Note: The [AudioWorkletGlobalScope](https://www.w3.org/TR/webaudio/#audioworkletglobalscope) may also contain any other data and code to be shared by these instances. As an example, multiple processors might share an ArrayBuffer defining a wavetable or an impulse response.